bluemira/equilibria/fem_fixed_boundary/fem_magnetostatic_2D.py

```diff
@@ -0,0 +1,180 @@
+            )  # define the boundaries
+        else:
+            self.boundaries = boundaries
+
+        # ======================================================================
+        # define the function space and bilinear forms
+        self.V = dolfin.FunctionSpace(self.mesh, "CG", p)  # the solution function space
+
+        # define trial and test functions
+        self.u = dolfin.TrialFunction(self.V)
+        self.v = dolfin.TestFunction(self.V)
+
+        # Define r
+        r = dolfin.Expression("x[0]", degree=p)
+
+        self.a = (
+            1
+            / (2.0 * dolfin.pi * 4 * dolfin.pi * 1e-7)
+            * (1 / r * dolfin.dot(dolfin.grad(self.u), dolfin.grad(self.v)))
+            * dolfin.dx
+        )
+
+        # initialize solution
+        self.psi = dolfin.Function(self.V)
+
+    def solve(
+        self,
+        g,
+        dirichlet_bc_function=None,
+        dirichlet_marker=None,
+        neumann_bc_function=None,
+    ):
+        """
+        Solve the weak formulation maxwell equation given a right hand side g,
+        Dirichlet and Neumann boundary conditions.
+
+        Parameters
+        ----------
+        g : dolfin.Expression or dolfin.Function
+            the right hand side function of the Poisson problem
+        dirichlet_bc_function : dolfin.Expression o dolfin.Function
+                              the Dirichlet boundary condition function
+        neumann_bc_function : dolfin.Expression or dolfin.Function
+                            the Neumann boundary condition function
+        dirichlet_marker : int
+                           the identification number for the dirichlet boundary
+
+        Returns
+        -------
+        psi : dolfin function
+            the poloidal magnetic flux as solution of the magnetostatic problem
+        """
+        if neumann_bc_function is None:
+            neumann_bc_function = dolfin.Expression("0.0", degree=2)
+
+        # define the right hand side
+        self.L = g * self.v * dolfin.dx - neumann_bc_function * self.v * dolfin.ds
+
+        # define the Dirichlet boundary conditions
+        if dirichlet_bc_function is None:
+            dirichlet_bc_function = dolfin.Expression("0.0", degree=2)
+            dirichlet_bc = dolfin.DirichletBC(
+                self.V, dirichlet_bc_function, "on_boundary"
+            )
+        else:
+            dirichlet_bc = dolfin.DirichletBC(
+                self.V, dirichlet_bc_function, self.boundaries, dirichlet_marker
+            )  # dirichlet_marker is the identification of Dirichlet BC in the mesh
+        bcs = [dirichlet_bc]
+
+        # solve the system taking into account the boundary conditions
+        dolfin.solve(self.a == self.L, self.psi, bcs)
+
+        # return the solution
+        return self.psi
+
+    def calculate_b(self):
+        """Calculates the magnetic field intensity from psi"""
+        w = dolfin.VectorFunctionSpace(
+            self.mesh, "P", 1
```

Comment:
What is "P" here? I couldn't find anything immediately apparent in the docs.

Reply:
Just took that code from the Fenics_tutorial (https://link.springer.com/book/10.1007/978-3-319-52462-7). I didn't realize that 'P' is not anymore explicitly used (even if still recognized), instead 'CG' should be used.
Changed and made a note with the reference to the Fenics tutorial.